core/src/test/scala/org/apache/spark/shuffle/unsafe/UnsafeShuffleSuite.scala

```diff
@@ -41,6 +41,20 @@ class UnsafeShuffleSuite extends ShuffleSuite with BeforeAndAfterAll {
     conf.set("spark.shuffle.memoryFraction", "0.5")
   }
 
+  override def multipleAttemptConfs: Seq[(String,SparkConf)] = {
+    val kryoAndNoSpillMove = conf.clone()
+      .set("spark.serializer", "org.apache.spark.serializer.KryoSerializer")
+      .set("spark.shuffle.unsafe.testing.allowSpillMove", "false")
+    val noCompression = kryoAndNoSpillMove.clone()
+      .set("spark.shuffle.compress", "false")
+      .set("spark.file.transferTo","false")
+    Seq(
+      "slow merge path" -> kryoAndNoSpillMove,
+      "filestream based fast merge" -> noCompression
+    )
```

Comment:
there should be another case here for the `transferTo` path as well, but that path fails with the msg about "Please check your kernel version to see if it is 2.6.32".  I don't understand the intricacies of when the channel's position is wrong from the discussion here https://issues.apache.org/jira/browse/SPARK-3948 so I just left it as is.

(which is another reason to consider an approach where the same files are written to, so append vs. overwrite doesn't matter)